### PR TITLE
Update docs

### PR DESCRIPTION
**Pull request by SHtress**

### CODE REVIEW

File: docs/dev/codestyle.md [hunk 1]
```diff
@@ -0,0 +1,309 @@
...
+SC_CHECK_LESS(a, b, ()); // equal to SC_CHECK(a < b, ());
+SC_CHECK_LESS_EQ(a, b, ()); // equal to SC_CHECK(a <= b, ());
+```
+
+- Logging:
+  * `SC_LOG_DEBUG(msg)` - print message in Debug builds. Prefix: `[Debug]`
+  * `SC_LOG_INFO(msg)`
+  * `SC_LOG_WARNING(msg)`
+  * `SC_LOG_ERROR(msg)`
+  * `SC_LOG_INFO_COLOR(msg, color)` - print colored info message. Look color constants in `ScConsole::Color`
```
Comment (MksmOrlov) [inline docs/dev/codestyle.md:289]: Добавь точку с запятой в конце

File: docs/dev/pr.md [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+
+Possible tags:
+
+  * `[test]` - changes in tests;
+  * `[review]` - commits with review fixes;
+  * `[refactor]` - commits with some code refactoring;
+  * `[changelog]` - use when you update changelog;
+  * `[doc]` - use when you update documentation;
+  * `[docker]` - changes in Dockerfile, .dockerignore or Docker image build pipeline
+  * `[scripts]` - updates in the `sc-machine/scripts` files
```
Comment (MksmOrlov) [inline docs/dev/pr.md:24]:
```suggestion
  * `[scripts]` - updates in the `scl-machine/scripts` files
```

File: docs/dev/pr.md [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+
+Each commit message should be formed as: `[tag1]...[tagN] Message text (#issue)`.
+
+Message text should start from an upper case letter. If commit doesn't fix or implement any #issue, then it shouldn't be pointed in commit message.
+
+Examples:
+<pre>
+[cpp] Colored log output
+[cpp][test] Add unit test for ScEvent class
+[kpm][search] Relation type check added
```
Comment (MksmOrlov) [inline docs/dev/pr.md:13]: there is no kpm and cpp

File: docs/dev/pr.md [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+</pre>
+
+Possible tags:
+
```
Comment (MksmOrlov) [inline docs/dev/pr.md:17]: Add smth as [inference] or [dia] [ria]. Tag to develop inference agents logic

**Review by MksmOrlov — CHANGES_REQUESTED** [review]
How to name branches?

**Review by MksmOrlov — APPROVED** [review]

File: docs/dev/git-workflow.md [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+To update your main branch from `upstream` use:
+```sh
+git fetch upstream
+git checkout main
+git merge upstream/main
+git push origin main
+```
+
+### Common rules
+  * use `git rebase` instead of `merge`
```
Comment (AlexandrZagorskiy) [inline docs/dev/git-workflow.md:22]: when I should use rebase instead of merge?

File: docs/dev/pr.md [hunk 6]
```diff
@@ -0,0 +1,86 @@
...
+Examples:
+<pre>
+[doc] Add documentation for direct inference agent
+[dia][test] Add unit tests
+</pre>
+
+Possible tags:
+
+  * `[dia]` - changes in direct inference agent;
+  * `[ria]` - changes in reverse inference agent; (or [iia] - changes in inverse/indirect inference)
```
Comment (AlexandrZagorskiy) [inline docs/dev/pr.md:18]: Better to remove any choices, this way will be more confusing

**Review by AlexandrZagorskiy — CHANGES_REQUESTED** [review]

File: docs/changelog.md [hunk 7]
```diff
@@ -7,8 +7,14 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
...
 
 ### Added
+- Add oportunity to generate and search in structure for DirectInferenceAgent
+- Add oportunity to only search and check formula, but not generate
+- Add mechanism for substituon values into variables
 - Documentation in scn-tex format
 - README
 - Changelog
+- Add pre-commit
+- Add clang format check
```
Comment (MksmOrlov) [inline docs/changelog.md:17]: move to the top of the "Added" section

**Review by MksmOrlov — APPROVED** [review]

**Review by AlexandrZagorskiy — APPROVED** [review]

**Pull request merged**